### ISSUE TITLE
‘dumb-jump-go’ is an obsolete command (as of 2020-06-26)

### ISSUE THREAD

**Issue by avatar-lavventura** [jacktasia/dumb-jump#365]
I was using dumb-jump (https://github.com/jacktasia/dumb-jump) but suddenly it stops working.

I am getting following error:

```
‘dumb-jump-go’ is an obsolete command (as of 2020-06-26); 
‘dumb-jump-go’ has been obsoleted by the xref interface.

’function’ python  declaration not found.
```

-------

My setup:
```
(add-hook 'xref-backend-functions #'dumb-jump-xref-activate)
(global-set-key "\C-x\C-o" 'dumb-jump-go-prefer-external-other-window);
(global-set-key "\C-x\C-k" 'xref-pop-marker-stack);
(global-set-key "\C-x\C-j" 'dumb-jump-go);
(global-set-key "\C-x\C-k" 'dumb-jump-back);
```

**Comment by jacktasia** [comment]
It should still be working even if it has been marked as obsolete. The real error here is `’function’ python  declaration not found.` which just means it could not find any results for whatever you were searching for.

```
def function():
    pass


function() # <-jumping from here worked for me.
```

**Comment by avatar-lavventura** [comment]
@jacktasia it also prevents me to use ‘dumb-jump-back’ and automatically function converted into ‘xref-pop-marker-stack’.

```
‘dumb-jump-back’ is an obsolete command (as of 2020-06-26); 
‘dumb-jump-back’ has been obsoleted by ‘xref-pop-marker-stack’.
```

**Comment by phikal** [comment]
What do you mean by "prevents"? Does it not jump back? Currently `dumb-jump-back` is not much more than a wrapper around `pop-tag-mark`, that in turn is just an alias for `xref-pop-marker-stack`, which is already bound to `M-.`. What happens when you press that key?

@jacktasia It might make sense to point that out in the deprecation string.

**Comment by avatar-lavventura** [comment]
I can jump back using ‘dumb-jump-back’ that is aliased to ‘xref-pop-marker-stack’ and `pop-tag-mark` works too (as I understand).

But I cannot use ‘dumb-jump-go’ but when I try `elpy-goto-definition` , which works. So maybe if `elpy-mode` is on ‘dumb-jump-go’ could be aliased to `elpy-goto-definition`.

**Comment by jacktasia** [comment]
@avatar-lavventura Seems like something else is going on here. Can you try [turning on debugging](https://github.com/jacktasia/dumb-jump#debugging-a-jump) and sharing the results from the `*Messages*` buffer here?

**Comment by kshartman** [comment]
Its also breaking my edit flow.  I will just fork it and remove the offending code.  I don't want the xref interface because. You should have an option to turn off the warnings and just do what it used to.

**Comment by phikal** [comment]
Yes, we added an user option (`dumb-jump-disable-obsolete-warnings` for anyone reading this thread in the future) to silence the legacy warnings. May I ask where you tried to find out how to silence the warnings? Would it have helped if the README mentioned it?

Also, on the Xref interface, the comment you cite was added over 5 years ago, and is negligible for all practical purposes. The interface is used far too widely at this point for any major changes to be made of a whim.  I might check with emacs-devel to see if that note could get removed.

**Comment by kshartman** [comment]
Guilty.  I didn't read the doc.  My first response is to grab the code and change it when I don't like it.  That is way I have always treated emacs since 1985.  The whole point is if you don't like it you can change it.  That is when I saw the variable.  And I still don't like xref.  I was forced to try it today because of the obsolete warning.  This code is way better for my workflow.  I really, really like it and i have it tuned for the work I primarily do now. Sorry for the excess verbiage.  I should have read more before commenting.  Thanks for doing this code.

**Comment by phikal** [comment]
>  My first response is to grab the code and change it when I don't like it

I can certainly relate to that.

> I will tell you why I despise the xref interface. Because it creates a buffer instead of a popup menu.

As you've said, Xref can be modified to use a different UI. There is a [posframe](https://github.com/fmdkdd/dotfiles/blob/master/emacs/.emacs.d/elisp/xref-posframe.el) based approach that used a child window to generate a popup (which is cleaner but more experimental than the classical popup approach that uses overlays AFAIK). 

There was some discussion around this topic on the mailing list a few months ago: https://lists.gnu.org/archive/html/emacs-devel/2021-01/msg00263.html, in case you are interested.

**Comment by dgutov** [comment]
@kshartman 

Try this:

```el
(setq xref-show-definitions-function #'xref-show-definitions-completing-read)
```

If you have any third-party completion framework enabled like Ivy or Helm, their UI should be used automatically.

**Comment by thisch** [comment]
Thx a lot for this information @dgutov. I've been looking for this config since I switched to the xref interface in dumb-jump. I think this should be mentioned in the README.

**Comment by phikal** [comment]
The only issue I see is that `xref-show-definitions-completing-read` hasn't been release yet, either as a core package or on ELPA.

**Comment by dgutov** [comment]
@phikal It's in release 1.1.0 now. :wink:

**Comment by avatar-lavventura** [comment]
@dgutov Should I close the issue? I assume its solved.

**Comment by phikal** [comment]
In some sense, the issue was already solved by #381, but your problem might have been something else, as implied in https://github.com/jacktasia/dumb-jump/issues/365#issuecomment-686597233.

**Comment by TrunovS** [comment]
For me see some strange behavior with ```xref``` backend which doesn't shows with obsolete ```dumb-jump-go```.
When working through Tramp, through ```xref``` jump to candidate goes in context of local machine, and opening local buffer as no such file exists.
The same jump to candidate through obsolete ```dumb-jump-go``` works just fine.

**Comment by digaoddc** [comment]
I configured a command with `CTRL+G+J` using xref to avoid this issue.

```
;; Configure go to definition command with alt+g+j
(use-package xref
:bind ("M-g j" . xref-find-definitions)
:config (setq xref 'helm)
:ensure)
```

**Comment by d99kris** [comment]
I was also facing this issue with `dumb-jump` on macOS. What solved the problem for me was a [blog post](https://martinfowler.com/articles/2023-xref-problem.html) suggestion to add this to emacs config:
```
(setq dumb-jump-force-searcher 'rg)
```
